File: Task-03/Ruby/subtask3.rb

puts "Enter the number of rows for the diamond (n): "  #input
n = gets.to_i

middle = (n / 2) + 1

# Generate the upper part
(1..middle).each do |i|
  print ' ' * (middle - i)
  print '*' * (2 * i - 1)
  puts
end

# Generate the lower part 
(middle-1).downto(1) do |i|
  print ' ' * (middle - i)
  print '*' * (2 * i - 1)
  puts
end

File: Task-03/ELixir/subtask3.exs
defmodule DiamondPattern do
  def run do
    IO.write("Enter the number of rows for the diamond (n): ")
    n = IO.gets("") |> String.trim() |> String.to_integer()

    middle = div(n + 1, 2)

    # Generate the upper part of the diamond
    Enum.each(1..middle, fn i ->
      spaces = middle - i
      stars = 2 * i - 1
      IO.puts("#{String.duplicate(" ", spaces)}#{String.duplicate("*", stars)}")
    end)

    # Generate the lower part of the diamond
    Enum.each((middle - 1)..1, fn i ->
      spaces = middle - i
      stars = 2 * i - 1
      IO.puts("#{String.duplicate(" ", spaces)}#{String.duplicate("*", stars)}")
    end)
  end
end

DiamondPattern.run()
